### PR TITLE
Support tensor arguments in clip.

### PR DESCRIPTION
**Pull request by jacquerie**
### Summary

I'm probably not doing this the right way, but I wanted to watch it fail on Travis (I'm sure I correctly installed only the Tensorflow backend locally).

### Related Issues

Closes https://github.com/keras-team/keras/issues/11435.

### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [x] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/backend/cntk_backend.py [hunk 1]
```diff
@@ -936,12 +936,23 @@ def pow(x, a):
...
+
+    # Arguments
+        x: Tensor or variable.
+        min_value: Python float, integer or tensor.
+        max_value: Python float, integer or tensor.
+
+    # Returns
+        A tensor.
+    """
+    if isinstance(min_value, (int, float)) and isinstance(max_value, (int, float)):
```
Comment (fchollet) [inline keras/backend/cntk_backend.py]: Please break line into 2, for length

```python
if (isinstance(min_value, (int, float)) and
    isinstance(max_value, (int, float))):
```

(same below)

File: tests/keras/backend/backend_test.py [hunk 2]
```diff
@@ -2035,6 +2035,15 @@ def test_variable_support_bool_dtype(self):
             with pytest.raises(TypeError):
                 K.variable('', dtype='unsupported')
 
+    def test_clip_supports_tensor_arguments(self):
+        # GitHub issue: 11435
+        x = K.variable([-10., -5., 0., 5., 10.])
+        min_value = K.variable([-5., -4., 0., 3., 5.])
+        max_value = K.variable([5., 4., 1., 4., 9.])
+
+        assert np.alltrue(K.eval(K.clip(x, min_value, max_value)) ==
```
Comment (fchollet) [inline tests/keras/backend/backend_test.py]: Use `assert_allclose`` instead

File: keras/backend/cntk_backend.py [hunk 3]
```diff
@@ -936,12 +936,24 @@ def pow(x, a):
...
-        max_value = min_value
-    if max_value is None:
-        max_value = np.inf
+    """Element-wise value clipping.
+
+    # Arguments
+        x: Tensor or variable.
+        min_value: Python float, integer or tensor.
+        max_value: Python float, integer or tensor.
+
```
Comment (gabrieldemarmiesse) [inline keras/backend/cntk_backend.py]: The theano and cntk backend functions don't need docstrings. The tensorflow backend docstrings are used for the documentation.

File: keras/backend/theano_backend.py [hunk 4]
```diff
@@ -694,8 +694,22 @@ def pow(x, a):
...
+    """Element-wise value clipping.
+
+    # Arguments
+        x: Tensor or variable.
+        min_value: Python float, integer or tensor.
+        max_value: Python float, integer or tensor.
+
+    # Returns
+        A tensor.
+    """
```
Comment (gabrieldemarmiesse) [inline keras/backend/theano_backend.py]: Same here.

File: keras/backend/cntk_backend.py (hunk 1 above)
Comment (gabrieldemarmiesse) [reply keras/backend/cntk_backend.py]: @jacquerie I believe this is the right way to do it.

**Pull request merged**